### PR TITLE
Week-5 - Json Converter implementation + test implementation

### PR DESCRIPTION
**Pull request by Locomody**

### CODE REVIEW

File: src/main/scala/karazin/scala/users/group/week5/homework/givens.scala [hunk 1]
```diff
@@ -6,12 +6,26 @@ object givens:
...
+    def encode(src: Int): String = src.toString
+
+  given BoolEncoder: JsonStringEncoder[Boolean] with
+    def encode(src: Boolean): String = src.toString
+
+  given StringEncoder: JsonStringEncoder[String] with
+    def encode(src: String): String = "\"" + src + "\""
+
+  given ListEncoder[T](using encoder: => JsonStringEncoder[T]): JsonStringEncoder[List[T]] with
+    def encode(src: List[T]): String = "[" + src.foldLeft(List[String]()) {(acc, src) => acc :+ encoder.encode(src)}.mkString(", ") + "]"
```
Comment (IgorWolkov) [inline src/main/scala/karazin/scala/users/group/week5/homework/givens.scala:24]: There is an extended version of `.mkString(", ")` `.mkString("[", ",", "]") so the fold can be simplified with map.
But it's not an issue.

**Review by IgorWolkov — CHANGES_REQUESTED** [review]
Looks like you have one failed test case.

File: src/test/scala/karazin/scala/users/group/week5/homework/GivensSuite.scala [hunk 2]
```diff
@@ -22,7 +24,91 @@ import scala.concurrent.Future
...
  */
 class GivensSuite extends munit.FunSuite:
-  
-  test("successful test example") {
-    assertEquals(42, 42)
+
+  test("Int even") {
+    assertEquals(
+      JsonStringEncoder[Int]
+        .encode(117), "200"
```
[the remark this answers is not in the document]
Comment (Locomody) [reply src/test/scala/karazin/scala/users/group/week5/homework/GivensSuite.scala:31]: oh sure, it fails)

**Comment by Locomody** [comment]
> 
Fixed

**Review by IgorWolkov — APPROVED** [review]
Good job.

**Pull request merged**